mne/viz/utils.py

```diff
@@ -2336,44 +2336,34 @@ def _setup_ax_spines(axes, vlines, tmin, tmax, invert_y=False,
         axes.yaxis.set_label_coords(-0.05, 1 - ylabel_height
                                     if invert_y else ylabel_height)
 
-    # this section is very long because the axes can become very small
-    # with topo plotting. This interferes with boundaries if the epoch is short
+    # the axes can become very small with topo plotting. This prevents the
+    # x-axis from shrinking to length zero if truncate_xaxis=True, by adding
+    # new ticks that are nice round numbers close to (but less extreme than)
+    # tmin and tmax
     vlines = [] if vlines is None else vlines
     xticks = sorted(list(set([x for x in axes.get_xticks()] + vlines)))
-    x_extrema = [t for t in xticks if tmax >= t and t >= tmin]
-    if len(x_extrema) < 2:  # can happen with one time point
-        x_extrema = [tmin, tmax]
-    if truncate_xaxis is True:
-        axes.spines['bottom'].set_bounds(x_extrema[0], x_extrema[-1])
+    ticks_in_range = [t for t in xticks if tmax >= t >= tmin]
+    if len(ticks_in_range) < 2:
+        def log_fix(tval):
+            exp = np.log10(np.abs(tval))
+            return np.sign(tval) * 10 ** (np.fix(exp) - (exp < 0))
+        tlims = np.array([tmin, tmax])
+        temp_ticks = log_fix(tlims)
+        closer_idx = np.argmin(np.abs(tlims - temp_ticks))
+        further_idx = np.argmax(np.abs(tlims - temp_ticks))
+        start_stop = [temp_ticks[closer_idx], tlims[further_idx]]
+        step = np.sign(np.diff(start_stop)) * np.max(np.abs(temp_ticks))
+        tts = np.arange(*start_stop, step)
+        ticks_in_range = sorted(ticks_in_range + [tts[0]] + [tts[-1]])
```

Comment:
you may want to modify this block; it always adds 2 additional ticks (so if there is 1 tick in range, you end up with 3; if zero, you end up with 2).  The ticks it adds will be as close as possible to tmin and tmax (without exceeding them) while still being an integer times a power of ten.

Reply:
I think this is probably ideal for do_topo, and I would call it iff do_topo.